src/core_plugins/vega/public/vega_view/vega_base_view.js

```diff
@@ -80,6 +67,31 @@ export class VegaBaseView {
     }
   }
 
+  createViewConfig() {
+    const config = {
+      logLevel: vega.Warn,
+      renderer: this._parser.renderer,
+    };
+
+    // Override URL sanitizer to prevent external data loading (if disabled)
+    const loader = vega.loader();
+    const originalSanitize = loader.sanitize.bind(loader);
+    loader.sanitize = (uri, options) => {
+      if (typeof uri === 'function') {
+        // If uri is a function, it must have come from this integration layer,
+        // because user can only supply pure JSON data structure.
+        // A function means that the value can be trusted (e.g. emsfile service)
```

Comment:
This is somewhat of a nit, but I wouldn't do this kind of param datatype overloading. I know JS allows it but that's because JS is awful.

Maybe add an option-param, "should_sanitize: [true|false]", or unpack the `uri` before passing it to sanitize (?)

Reply:
@thomasneirynck I cannot use any kind of a regular data because in that case it could be added by the user via JSON.  It has to be something that's impossible to generate via JSON.   Which means that even if I pass it as an object, I would still have the same check: `if (typeof uri.should_sanitize === 'function') {...}`.  I think this is worse, because the intention is even less clear.

Reply:
ic. I meant adding it to the 2nd options param. but yeah, it just came across as wonky, but maybe it is indeed the easiest way to do this.

Reply:
I would also suggest using something on the `options` param if that is possible? Unfortunately I lack again a Vega Spec in this PR to properly test it locally if we can pass data to that, and the Vega documentation of that method doesn't make it clear to me, where this `options` parameter come from exactly :-) It would be super awesome if we could attach a Vega spec to any Vega PR, that will reproduce the scenario to test in this PR.

Also if we need a non JSON compatible datatype, I would rather work with `Symbol` here, since this is kind of the proper use-case for it - creating non recreatable tokens. I would export a method from this file `bypassExternalUrlCheck(uri)`, that will be called in the EMS file , and pass in the URL. The implementation could look a bit like the following (if we can't leverage the `options` parameter to the loader somehow).

*vega_base_view.js*
```js
const bypassToken = Symbol();

export function bypassExternalUrlCheck(url) {
  return { url, bypassToken };
}

// ...
loader.sanitize = (uri, options) => {
  if (uri.bypassToken && uri.bypassToken === bypassToken) {
     uri = uri.url;  
  } else if // ...
};
```

That way only code with access to that function is able to sanitize URLs for bypassing, and it's also a bit more explicit, by encapsulating it in its own method, instead of relying for all "consumers" (like the EMS service) to know specifics about the implementation of this check (in this case, that they would need to use a method).

Reply:
The `options` param is the same per entire graph, not per data source. The only per-data-source param is the `uri`, which is simply a reference to the user-submitted `uri` value in the graph's data source.

@timroes good point about the test spec, thx, will do.

I like your symbol idea too, will do.

Reply:
Thanks for clarifying what the `options` param does. In this case I would go for a solution like sketched above :)